### PR TITLE
EDUVPNMAC-2: Proof of concept

### PR DESCRIPTION
**Pull request by johankool**

### CODE REVIEW

File: EduVPN/ChooseProviderViewController.swift [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    func numberOfRows(in tableView: NSTableView) -> Int {
+        return providers.count
+    }
+    
+}
+
+extension ChooseProviderViewController: NSTableViewDelegate {
+    
+    func tableView(_ tableView: NSTableView, willDisplayCell cell: Any, for tableColumn: NSTableColumn?, row: Int) {
+        (cell as! NSTableCellView).textField?.stringValue = providers[row].displayName
```
Comment (petercv) [inline EduVPN/ChooseProviderViewController.swift]: Why don't you use if let ... constructions?

File: EduVPN/ChooseProviderViewController.swift [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    }
+    
+    func tableViewSelectionDidChange(_ notification: Notification) {
+        (view.window?.windowController as? MainWindowController)?.showAuthenticating()
```
Comment (petercv) [inline EduVPN/ChooseProviderViewController.swift]: Same here.

File: EduVPN/ConnectionService.swift [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+        try config.write(to: fileURL, atomically: true, encoding: .utf8)
+        return fileURL
+    }
+    
+    private var connection: NSXPCConnection?
+    
+    private var authRef: AuthorizationRef? = nil
+    private var authorization: Data!
+    
+//    private func connectToAuthorization() {
```
Comment (petercv) [inline EduVPN/ConnectionService.swift]: cocwoc

**Comment by johankool** [comment]
Merging this one despite comments so we have at least commit with proof of concept in repo. Afaict comments are address in next PR.

**Pull request merged**